test/urns.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @see https://en.wikipedia.org/wiki/Uniform_Resource_Name
+ */
+export const correctUrns = [
+  'urn:isbn:0451450523',
+  'urn:isan:0000-0000-2CEA-0000-1-0000-0000-Y',
+  'urn:ISSN:0167-6423',
+  'urn:ietf:rfc:2648',
+  'urn:mpeg:mpeg7:schema:2001',
+  'urn:oid:2.16.840',
+  'urn:uuid:6e8bc430-9c3a-11d9-9669-0800200c9a66',
+  'urn:nbn:de:bvb:19-146642',
+  'urn:lex:eu:council:directive:2010-03-09;2010-19-UE',
+  'urn:lsid:zoobank.org:pub:CDC8D258-8F57-41DC-B560-247E17D3DC8C',
+  'urn:epc:class:lgtin:4012345.012345.998877',
+  'urn:epc:id:sgtin:0614141.112345.400',
+  'urn:epc:id:sscc:0614141.1234567890',
+  'urn:epc:id:sgln:0614141.12345.400',
+  'urn:epc:id:bic:CSQU3054383',
+  'urn:epc:id:imovn:9176187',
+  'urn:epc:id:gdti:0614141.12345.400',
+  'urn:mrn:iala:aton:us:1234.5',
+  'urn:mrn:iala:vts:ca:ecareg',
+  'urn:mrn:iala:wwy:us:atl:chba:potri',
+  'urn:mrn:iala:pub:g1143',
+  'urn:microsoft:adfs:claimsxray',
+  'urn:isan:0000-0000-9E59-0000-O-0000-0000-2',
+];
+
+export const incorrectUrns = [
```

Comment:
Where did you find these?
I'm not sure why these would be invalid.

Reply:
The [spec rfc2141](https://datatracker.ietf.org/doc/html/rfc2141)
Gives the following format:
`<URN> ::= "urn:" <NID> ":" <NSS>`
AWS (among others) has increased the number parts of the URN.

I added the source where those came from.

Reply:
`<NSS>` actually also accepts `:` characters, so AFAIK these URNs are valid.

Reply:
Okay, I saw it in the spec that NSS may have :
The reference: https://datatracker.ietf.org/doc/html/rfc3986#section-3.3

Will look into it. I thought I made the conclusion because of another lib that had limitations on parts count of the string split by :